src/main.rs

```diff
@@ -345,17 +345,15 @@ impl App {
             }
         }
 
-        let run_command: Command = command_from_argument(args.get_one::<String>("command"))?
+        let mut run_command = command_from_argument(args.get_one::<String>("command"))?
             .expect("--command is required to run solution.");
 
-        let timeout_seconds: f64 = *args.get_one::<f64>("timeout").unwrap_or(&5.0);
-
-        let timeout = std::time::Duration::from_micros(match timeout_seconds {
-            x if x.is_nan() => return Err(anyhow!("Timeout can't be NaN")),
-            x if x < 0.0 => return Err(anyhow!("Timeout can't be negative (use 0 for no timeout)")),
-            x if x == 0.0 => u64::MAX,
-            x => (1e6 * x) as u64,
-        });
+        let timeout = match *args.get_one::<f64>("timeout").unwrap_or(&5.0) {
+            secs if secs.is_nan() => return Err(anyhow!("Timeout can't be NaN")),
+            secs if secs < 0.0 => return Err(anyhow!("Timeout can't be negative (use 0 for no timeout)")),
+            secs if secs == 0.0 => std::time::Duration::MAX,
+            secs => std::time::Duration::from_secs(secs as u64),
```

Comment:
This is a change in behavior. The code used to able to handle decimals like `--timeout 0.2` but this would round that down to 0.

Reply:
I'm an idiot